internal/proxy/oauthproxy.go

```diff
@@ -757,9 +803,11 @@ func (p *OAuthProxy) Authenticate(rw http.ResponseWriter, req *http.Request) (er
 	} else if session.ValidationPeriodExpired() {
 		// Validation period has expired, this is the shortest interval we use to
 		// check for valid requests. This should be set to something like a minute.
-		// This calls up the provider chain to validate this user is still active
-		// and hasn't been de-authorized.
-		ok := p.provider.ValidateSessionState(session, allowedGroups)
+		// In this case we:
+		// - call up the provider chain to validate this user is still active and hasn't been de-authorized.
+		// - run any defined email domain, email address, and email group validators against the session
+
+		ok := p.provider.ValidateSessionToken(session)
```

Comment:
This refactoring also makes me wonder if this function signature should be the same as refresh? Presumably the validation of the session token could fail in the same two ways:

1. We just are physically unable to validate the session token (i.e. it error'd)
2. We successfully validated the session but the user is no longer authorized (bool is false)

Let's make a note that we can clean this up in a following PR? Overall a good set of refactors!